proto/authn/v1/authn.proto

```diff
@@ -0,0 +1,71 @@
+// Authn protocol
+syntax = "proto3";
+package xmtp.authn.v1;
+
+import "google/api/annotations.proto";
+
+option go_package = "./authn/v1";
+
+// ----------------------------------------------------------------------------
+// These objects are copied from xmtpjs.
+// Compile failure occurs with GoGo protobuf as optional fields are not 
+// supported.
+// Migrating to go
+// ----------------------------------------------------------------------------
+
+// Signature represents a generalized public key signature,
+// defined as a union to support cryptographic algorithm agility.
+message Signature {
+    message ECDSACompact {
+        bytes bytes = 1; // compact representation [ R || S ], 64 bytes
+        uint32 recovery = 2; // recovery bit
+    }
+    oneof union {
+        ECDSACompact ecdsa_compact = 1;
+    }
+}
+
+message Secp256k1Uncompresed {
+    // uncompressed point with prefix (0x04) [ P || X || Y ], 65 bytes
+    bytes bytes = 1; 
+}
+// PublicKey represents a generalized public key,
+// defined as a union to support cryptographic algorithm agility.
+message PublicKey {
+    uint64 timestamp = 1;
+    Signature signature = 2;
+    oneof union {
+        Secp256k1Uncompresed secp256k1_uncompressed = 3;
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+message ClientAuthRequest {
+    PublicKey identity_key = 1;
+    bytes auth_data_bytes = 3;
+    Signature auth_data_signature = 4;
+}
+  
+message ClientAuthResponse {
+    oneof union {
+        bytes token = 1;  // if auth was successful
+        string error_str = 2; // if auth was not successful
+    }
+}
+  
+message AuthData {
```

Comment:
One thing we are going to have to figure out is how we consistently serialize this `AuthData` into `auth_data_bytes` between the client and server, now that we only have a JSON API and no client-side protobuf. Whatever gets serialized into `auth_data_bytes` needs to be verifiable by the server. I guess we could use actual-protobuf just for this step, but that feels awkward.

Reply:
I think protobuf makes sense since we are using protobuf for messages themselves.

Reply:
I'm fine with using `protobuf`. Keeps things leaner, which matters when you are stuffing it into a HTTP header.

Wouldn't you have to include the full `ClientAuthRequest` into the header to do the real validation and tie it to a wallet address? Not just the `AuthData`, but the pub key and wallet signature.

Reply:
Sorry, yes, the idenity key and the signatures would have to be part of it.

Reply:
If we are going with this suggestion (which I'm into assuming Jazz doesn't have any concerns), then we would want to do the codegen here with `protoc-gen-ts_proto` just like we do in the `buf.gen.yaml` on the client.